opt/scopeserver/dataserver/modules/gserver/GServer.py

```diff
@@ -987,6 +988,65 @@ def loomUploaded(self, request, content):
         uploadedLooms[request.UUID].add(request.filename)
         return s_pb2.LoomUploadedReply()
 
+    def get_cluster_marker_table(self, loomFilePath, clusteringID, clusterID) -> pd.DataFrame:
+        loom = self.lfh.get_loom(loom_file_path=loomFilePath)
+
+        def create_cluster_marker_metric(metric):
+            return loom.get_cluster_marker_metrics(
+                clustering_id=clusteringID, cluster_id=clusterID, metric_accessor=metric["accessor"]
+            )
+
+        def merge_cluster_marker_metrics(metrics):
+            return functools.reduce(
+                lambda left, right: pd.merge(left, right, left_index=True, right_index=True, how="outer"), metrics,
+            )
+
+        md_clustering = loom.get_meta_data_clustering_by_id(id=clusteringID, secret=self.config["dataHashSecret"])
+        md_cmm = md_clustering["clusterMarkerMetrics"]
+        cluster_marker_metrics = merge_cluster_marker_metrics(metrics=[create_cluster_marker_metric(x) for x in md_cmm])
+        # Keep only non-zeros elements
+        nonan_marker_mask = cluster_marker_metrics.apply(
+            lambda x: functools.reduce(np.logical_and, ~np.isnan(x)), axis=1
+        )
+        return cluster_marker_metrics[nonan_marker_mask]
+
+    def getGProfilerLink(self, request, context):
+        cluster_marker_table = self.get_cluster_marker_table(
+            loomFilePath=request.loomFilePath, clusteringID=request.clusteringID, clusterID=request.clusterID
+        )
+        if "avg_logFC" not in cluster_marker_table.columns:
+            return s_pb2.GProfilerLinkOutReply(url="Error")
```

Comment:
In some of the later functions, I've been using a (Response: bool, Message: str) format to return any errors, and whether the function worked. It might be nice to do the same here, then you can return a meaningful error message to the user. 

Either way it would be nice to log an error on the backend here.

Reply:
I was lazy here :p but indeed better to have some structure for returning an error